src/libsync/creds/keychainchunk.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) by Michael Schuster <michael@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+#ifndef KEYCHAINCHUNK_H
+#define KEYCHAINCHUNK_H
+
+#include <QObject>
+#include <keychain.h>
+#include "accountfwd.h"
+
+// We don't support insecure fallback
+// #define KEYCHAINCHUNK_ENABLE_INSECURE_FALLBACK
+
+namespace OCC {
+
+namespace KeychainChunk {
+
+/*
+* Workaround for Windows:
+*
+* Split the keychain entry's data into chunks of 2048 bytes,
+* to allow 4k (4096 bit) keys / large certs to be saved (see limits in webflowcredentials.h)
+*/
+static constexpr int ChunkSize = 2048;
+static constexpr int MaxChunks = 10;
+
+/*
+ * @brief: Abstract base class for KeychainChunk jobs.
```

Comment:
Turns out it's not abstract... but now I also realize...
`start()` should be pure virtual in here, and likely we want to factor out all the `exec()` implementations in subclasses (which are somewhat close) into ideally a non-virtual `exec()` here (or at worst virtual to put the subclasses specific code in there).

Reply:
Back then I decided against fancy virtualization / subclassing to keep the interface stupid simple, mostly to avoid any casting mistakes in the keychain slots in `WebFlowCredentials` and to ease resulting own code readability there.

Reason: In the past someone confused `Read` and `Write` in the previous implementation, likely because of copy-pasting the generic `Job` cast.
This is why I chose for duplication in `KeychainChunk` instead, while heavily reducing it in `WebFlowCredentials`.

And yeah of course time was also an issue (surprise^^).

Any ideas? ;)

Reply:
I wouldn't call that fancy. Especially when it avoids having three almost identical but slightly different methods of the same name in subclasses. :-)

They're so close to be substitutable it's sad not to make them as such.

Reply:
Beside: I wish we had a proper Job interface... we got plenty of disjointed job families. Maybe one day I'll look at the feasibility of pull kcoreaddons to get KJob. One day...

Reply:
> I wouldn't call that fancy. Especially when it avoids having three almost identical but slightly different methods of the same name in subclasses. :-)

Fancy in the sense of anti-duplication ;-)

> 
> They're so close to be substitutable it's sad not to make them as such.

Yepp, I know... ^^